lib/rewrite/rewrite-subst.c

```diff
@@ -62,8 +62,27 @@ log_rewrite_subst_process(LogRewrite *s, LogMessage **pmsg, const LogPathOptions
                                   &new_length);
   if (new_value)
     {
+      msg_trace("Performing subst() rewrite",
+                evt_tag_str("rule", s->name),
+                evt_tag_str("value", log_msg_get_value_name(s->value_handle, NULL)),
+                evt_tag_printf("input", "%.*s", (gint) length, value),
+                evt_tag_str("type", self->matcher_options.type),
+                evt_tag_str("pattern", self->matcher->pattern),
+                evt_tag_str("replacement", self->replacement->template),
+                log_pipe_location_tag(&s->super));
       log_msg_set_value(*pmsg, self->super.value_handle, new_value, new_length);
     }
+  else
+    {
+      msg_trace("Performing subst() rewrite failed, pattern did not match",
+                evt_tag_str("rule", s->name),
+                evt_tag_str("value", log_msg_get_value_name(s->value_handle, NULL)),
+                evt_tag_printf("input", "%.*s", (gint) length, value),
+                evt_tag_str("type", self->matcher_options.type),
```

Comment:
optional: I would try to avoid duplication. A new function might be suitable: `rewrite_subst_trace(const gchar *str, LogRewriteSubst *self, gint length, const gchar *value);`

Reply:
I prefer that log messages are trivial to find using a simple grep, and also I find that arguments to a specific log message is a concept that might easily change and although it seems like code duplication, the concepts should be handled separately.

Also, this is not a pattern we do in general, and I am a bit opposed to it, if we did.

With that said I wouldn't do this now.